### DISCUSSION TITLE
MAUI with CSLA 5.x

### DISCUSSION THREAD

**Discussion by MGMaciej** [MarimerLLC/csla#4231] [Questions] [answered]
Hi,
Has anyone ever run a Maui N-Tier application with CSLA 5.x? I am trying to switch from Xamarin to Maui in a project where I'm still using CSLA 5.5.4, and I’m encountering the same problem as Sylum [here ](https://github.com/MarimerLLC/csla/discussions/3325) (Unable to read beyond the end of the stream).

I see the workaround from @rockfordlhotka [here](https://github.com/MarimerLLC/csla/issues/3791#issuecomment-2065381572), but it’s not working for me. I wonder if this is because I’m still using CSLA 5.x?

I think MAUI is newer than CSLA 5.x, so that might be the issue, but I’m trying to confirm if updating CSLA is the only way to resolve this.

**Comment by rockfordlhotka** [comment]
I don't know if v5 will work with MAUI, but I suspect that it will.

In the end, the missing piece in HttpProxy is to set the _content_ type. You should be able to to do that in v5.

**Comment by MGMaciej** [comment, marked as answer]
@rockfordlhotka , thank you for the quick reply. I had been struggling with this setup for a while. After your response, I decided to give it one more try, and this time, I was finally successful.

Here is what helped in my case:
```UseTextSerialization = true``` on the server controller...

```
[Route("api/[controller]")]
[ApiController]
public class DataPortalController : Csla.Server.Hosts.HttpPortalController
{
    public DataPortalController()
    {
        // Do not enable this only on server!
        UseTextSerialization = true;
    }

    [HttpGet]
    public string Get() => "DataPortal is running from MAUI";
}
```


And ``` UseTextSerialization = true ``` in the MauiProgram...

```     public static class MauiProgram
    {
        public static MauiApp CreateMauiApp()
        {
            var builder = MauiApp.CreateBuilder();
            builder
                .UseMauiApp<App>()
                .ConfigureFonts(fonts =>
                {
                    fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
                    fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
                });
            
            Csla.ApplicationContext.DataPortalProxy = typeof(Csla.DataPortalClient.HttpProxy).AssemblyQualifiedName;
            Csla.DataPortalClient.HttpProxy.UseTextSerialization = true;            
            Csla.ApplicationContext.DataPortalUrlString = "myurl";
            return builder.Build();
        }
    } 
```

#CSLA 5.5.4

**Reply by rockfordlhotka** [reply]
It is true that the client and server _both_ need to either use text serialization or not. So if the client is using text serialization then the server must do so as well. Text serialization might have been the default for a Xamarin client in CSLA 5.

**Comment by richardbartley** [comment]
I'm having some success with MAUI .Net8 and CSLA, but fell foul of trimming targeting Android in our DevOps pipeline and it would not build.  I'd prototyped it with a simple NVList Csla object and it worked, but then as I added other types it started throwing build errors.
There are things that the IL Linker cannot work out, certain code techniques that it cannot determine, reflection.
 
Various errors such as 
`ILLink : error IL1012: IL Trimmer has encountered an unexpected error.`
or
`error NETSDK1144: Optimizing assemblies for size failed. Optimization can be disabled by setting the PublishTrimmed property to false`

Managed to solve it by marking my business library project that has all my Csla objects with;
`<IsTrimmable>false</IsTrimmable>`

Maybe that helps someone.

**Reply by rockfordlhotka** [reply]
I suspect that is because the data portal operation methods _appear_ to be unused - they are private and never invoked directly, only dynamically via the data portal.

If they are trimmed away, obviously things can't work.
